source/vof_initial_conditions/function.cc

```diff
@@ -0,0 +1,132 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/vof/handler.h>
+#include <aspect/vof_initial_conditions/function.h>
+#include <aspect/postprocess/interface.h>
+
+namespace aspect
+{
+  namespace VoFInitialConditions
+  {
+    template <int dim>
+    Function<dim>::Function ()
+      :
+      n_init_samples (3)
+    {}
+
+    template <int dim>
+    unsigned int Function<dim>::n_samples () const
+    {
+      return n_init_samples;
+    }
+
+    template <int dim>
+    typename VoFInitType::Kind Function<dim>::init_type () const
+    {
+      return function_init_type;
+    }
+
+    template <int dim>
+    double
+    Function<dim>::
+    initial_value (const Point<dim> &position, const unsigned int n_field) const
+    {
+      return function->value(position, n_field);
+    }
+
+    template <int dim>
+    void
+    Function<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial Volume of Fluid model");
+      {
+        prm.enter_subsection("Function");
+        {
+          prm.declare_entry("Signed distance init", "false",
+                            Patterns::Bool (),
+                            "When set to true, initialization will be assumed to be a"
+                            "signed distance level set function.");
+
+          prm.declare_entry ("Number initialization samples", "3",
```

Comment:
Move this into VoFHandler, and remove the above parameter. All existing InitialComposition plugins can be used for the initialization samples method, and all new plugins (derived from VoFInitialComposition) will implement the signed distance function.